pkg/plugin/v3/scaffolds/internal/templates/main.go

```diff
@@ -213,13 +214,14 @@ func init() {
 }
 
 func main() {
+{{- if not .ComponentConfig }}
 	var metricsAddr string
-	var enableLeaderElection bool
 	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
+{{ end }}
+	var enableLeaderElection bool
 	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
```

Comment:
Since leader election is configured by the config, wouldn't its flag not be scaffolded if using a config patch?

Reply:
Great question, if possible yes, the difficult thing is we'd have to patch this before the configmap is written and I'm not aware right now of a way to do this or else `make run` will fail because the config file won't load. My first pass created separate files for each local vs in-cluster but I fear that could be confusing to folks.

Any chance you can think of a good way around that?